source/common/ssl/connection_impl.cc

```diff
@@ -251,8 +251,21 @@ std::string ConnectionImpl::subjectPeerCertificate() {
   if (!cert) {
     return "";
   }
-  bssl::UniquePtr<char> buf(X509_NAME_oneline(X509_get_subject_name(cert.get()), nullptr, 0));
-  return std::string(buf.get());
+
+  // Format settings for a single line in RFC 2253 format.
+  constexpr int indent = 0;
+  constexpr unsigned long flags = XN_FLAG_RFC2253;
```

Comment:
`XN_FLAG_RFC2253` is the only option that the OpenSSL documentation states is compatible with RFC 2253. The others are noted as "more readable" or "multiline", and I can't tell if they're compatible with any particular RFC or are just formats that the OpenSSL authors like.